### PR TITLE
Set maxFeePerGas to gasCap

### LINKED ISSUE #1874: Dynamically calculate MAX_FEE_FACTOR
Inspired by https://github.com/cowprotocol/services/pull/1867, constant `MAX_FEE_FACTOR` should be dynamically calculated based on the configuration parameters for solving time and submission deadline, since those can change.

### PR DESCRIPTION
**Pull request by fleupold**
# Description
Our gas estimation logic was written prior to EIP1559 which changed Ethereum's transaction fee mechanism from a "pay as bid" to a "truthful bidding" mechanism. Before, users would need to think

1. How much they are maximally willing to pay for their transaction (1000 gwei by default for us)
2. How much they should bid for transaction inclusion (dependent on current demand/supply)

They would always get charged 2.

Now, people only have to answer 1. and can rely on Ethereum's elastic base fee to make sure they are only getting charged as much as needed (the current base fee).

Priority fees remain a "pay as bid" part of the pricing, however those values tend to be very small (1-3 gwei) compared to the base fee (20-500 gwei).

This PR start to bid truthfully with regards to the maximum fee we are willing to pay. This simplifies a bunch of the solution submission logic as we now longer have to predict the maximum base fee in the future.

# Changes

- [x] Always set `maxFeePerGas` to `gas_price_cap`

## How to test
Run e2e test, see settlements are now submitted with very high cap.

## Related Issues

Fixes #1874

### CODE REVIEW

File: crates/driver/src/infra/blockchain/gas.rs [hunk 1]
```diff
@@ -42,9 +43,16 @@ impl GasPriceEstimator {
                     mempool.additional_tip_percentage,
                 )
             });
+        // Use the lowest max_fee_per_gas of all mempools as the max_fee_per_gas
+        let max_fee_per_gas = mempools
+            .iter()
+            .map(|mempool| eth::U256::from_f64_lossy(mempool.gas_price_cap))
```
Comment (sunce86) [inline crates/driver/src/infra/blockchain/gas.rs:49]: `gas_price_cap` should be probably updated to `U256`, not related to this PR

**Review by sunce86 — APPROVED** [review]
Are we gonna cleanup MAX_FEE_FACTOR in a follow up PR?

**Comment by fleupold** [comment]
> Are we gonna cleanup MAX_FEE_FACTOR in a follow up PR?

It's being remove as part of #2294

File: crates/driver/src/infra/blockchain/gas.rs (hunk 1 above)
Comment (fleupold) [reply crates/driver/src/infra/blockchain/gas.rs:49]: #2301

File: crates/driver/src/infra/blockchain/gas.rs [hunk 2]
```diff
@@ -42,9 +43,16 @@ impl GasPriceEstimator {
                     mempool.additional_tip_percentage,
                 )
             });
+        // Use the lowest max_fee_per_gas of all mempools as the max_fee_per_gas
```
Comment (MartinquaXD) [inline crates/driver/src/infra/blockchain/gas.rs:46]: Either the comment is wrong or you should be using `.min()`, no?
```suggestion
        // Use the highest max_fee_per_gas of all mempools as the max_fee_per_gas
```

**Review by MartinquaXD — APPROVED** [review]

**Pull request merged**